compilerProvider.js

```diff
@@ -0,0 +1,285 @@
+const path = require('path');
+const fs = require('fs');
+const request = require('request-promise');
+const requireFromString = require('require-from-string');
+const mkdirp = require('mkdirp');
+
+
+//------------------------------ Constructor/Config ------------------------------------------------
+
+/**
+ * Constructor:
+ * @param {Object} _config (see `provider.config`)
+ */
+function CompilerProvider(_config){
+  _config = _config || {};
+  this.config = Object.assign(this.config, _config);
+}
+
+/**
+ * Default configuration
+ * @type {Object}
+ */
+CompilerProvider.prototype.config = {
+  solc: null,
+  versionsUrl: 'https://solc-bin.ethereum.org/bin/list.json',
+  compilerUrl: 'https://solc-bin.ethereum.org/bin/',
+  compilerNpm: 'solc',
+  cache: true,
+  cachePath: '/var/lib/truffle/cache/solc/',
+}
```

Comment:
Set the default solc path at `var/lib`. Not sure about this - another candidate is `var/tmp`.
+ [link](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard) to file system hierarchy wiki
+ [link](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05.html) to var heirarchy linux doc.

Reply:
@gnidan noted this isn't cross-platform. There's a node-based [solution](https://github.com/avajs/find-cache-dir) supported by nyc/AVA that finds and creates application specific caches at `node_modules`. Seems like this should work for both local and global installs, and just makes more sense generally.